src/probabilities.jl

```diff
@@ -190,3 +198,51 @@ it may be explicitly extended for better performance.
 function outcomes(est::ProbabilitiesEstimator, x)
     return probabilities_and_outcomes(est, x)[2]
 end
+
+###########################################################################################
+# All probabilities
+###########################################################################################
+"""
+    allprobabilities(est::ProbabilitiesEstimator, x::Array_or_Dataset) → p
+
+Same as [`probabilities`](@ref), however ensures that outcomes with `0` probability
+are explicitly added in the returned vector. This means that `p[i]` is the probability
+of `ospace[i]`, with `ospace = `[`outcome_space`](@ref)`(est, x)`.
+
+This function is useful in cases where one wants to compare the probability mass functions
+of two different input data `x, y` under the same estimator. E.g., to compute the
+KL-divergence of the two PMFs assumes that the obey the same indexing. This is
+not true for [`probabilities`](@ref) even with the same `est`, due to the skipping
+of 0 entries, but it is true for [`allprobabilities`](@ref).
+"""
+function allprobabilities(est::ProbabilitiesEstimator, x::Array_or_Dataset)
+    probs, outs = probabilities_and_outcomes(est, x)
+    ospace = vec(outcome_space(est, x))
+    # We first utilize that the outcome space is sorted and sort probabilities
```

Comment:
Is the outcome space guaranteed to be sorted, though, or is the "sorting" just that we have happened to define the outcome space for the existing estimators such that they are sorted (according to *some* non-specified algorithm)?